cuda_core/cuda/core/experimental/_memory.py

```diff
@@ -293,3 +293,34 @@ def is_host_accessible(self) -> bool:
     @property
     def device_id(self) -> int:
         raise RuntimeError("the pinned memory resource is not bound to any GPU")
+
+
+class _SynchronousMemoryResource(MemoryResource):
+    __slots__ = ("_dev_id",)
+
+    def __init__(self, dev_id):
+        self._handle = None
+        self._dev_id = dev_id
+
+    def allocate(self, size, stream=None) -> Buffer:
+        if stream is None:
+            stream = default_stream()
+        ptr = handle_return(cuda.cuMemAlloc(size, stream._handle))
```

Comment:
I don't think this works? `cuMemAlloc` only takes 1 argument (so we don't need `stream` here). Could you test locally?

Reply:
you're right, I switched it and crequested a new t4 tesla lease to test it on colossus. The lease is for today so I will have a final draft ready soon. 

I assume it will work, but for sanity, and since we wont have a tcc device in the CI, I figured it was best to do a proper test and document the result